docs/packages/pkg/Boost.rst

```diff
@@ -75,21 +75,48 @@ Compatibility mode
 CMake options
 -------------
 
-CMake options can be passed to boost build using ``CMAKE_ARGS`` feature
+You can use ``CMAKE_ARGS`` feature
 (see
-`customization <https://github.com/ruslo/hunter/wiki/example.custom.config.id#custom-cmake-options>`__).
-Options of special form ``<COMPONENT-UPPERCASE>_<OPTION>=<VALUE>`` will
-be added to ``b2`` as ``-s <OPTION>=<VALUE>`` while building component .
-For example:
+`customization <https://github.com/ruslo/hunter/wiki/example.custom.config.id#custom-cmake-options>`__)
+to pass options to boost build or to append config macros in the default boost user
+config file (boost/config/user.hpp):
 
-.. code-block:: cmake
+- Options of special form ``<COMPONENT-UPPERCASE>_<OPTION>=<VALUE>`` will
+  be added to ``b2`` as ``-s <OPTION>=<VALUE>`` while building component .
+  For example:
+
+  .. code-block:: cmake
 
     hunter_config(Boost VERSION ... CMAKE_ARGS IOSTREAMS_NO_BZIP2=1)
     # add NO_BZIP2=1 to the b2 build of iostreams library, i.e. `b2 -s NO_BZIP2=1`
 
 -  `boost.iostreams
    options <http://www.boost.org/doc/libs/1_57_0/libs/iostreams/doc/index.html?path=7>`__
 
+- Options ``CONFIG_MACRO_<ID>=<VALUE>`` will append ``#define <ID> <VALUE>``
```

Comment:
What is the official Boost way to introduce new macro? Is it possible to use `b2 -s`?

Reply:
It is possible to pass macro when building boost but with `b2 define=MACRO_ID=value` or `b2 define=MACRO_ID`, not with `b2 -s JAM_VARIABLE=VALUE`, see http://www.boost.org/build/doc/html/bbv2/overview/invocation.html (as you know, jam variable are used by some boost libs to setup some build time only configurations, like libs dependencies)

I don't think there is more official way to introduce macro, via `define` build option or via `boost/config/user.hpp` (http://www.boost.org/doc/libs/1_65_1/libs/config/doc/html/index.html#boost_config.configuring_boost_for_your_platform.user_settable_options). If we use build time `define` then we have to add those macro using `target_compile_definition()` in `BoostConfig.cmake`. If we append to `boost/config/user.hpp` that macros are in effect when building boost and for projects that use boost even for projects that don't use cmake/hunter (maybe a 3rd party dependency).

I have doubts about which of two approach is better but I think append to `boost/config/user.hpp` is simple and convenient. What do you think?

Reply:
> as you know, jam variable are used by some boost libs ...

My knowledge of Boost build system is quite limited :)

> for projects that use boost even for projects that don't use cmake/hunter (maybe a 3rd party dependency)

In general it's better not to rely on that. We already adding `BOOST_ALL_NO_LIB` and `BOOST_ALL_DYN_LINK`:
https://github.com/ruslo/hunter/blob/538281f5d49c4337d36d8af900f5481038fdad68/cmake/templates/BoostConfig.cmake.in#L53-L66

There may be more in future. Also it may work for you for Boost but when you decide to add more packages it will break.

> I have doubts about which of two approach is better but I think append to boost/config/user.hpp is simple and convenient. What do you think?

I'm okay with it if it will work correctly.